controllers/mysqlcluster_controller_test.go

```diff
@@ -1612,4 +1613,264 @@ var _ = Describe("MySQLCluster reconciler", func() {
+		sts.Status.UpdateRevision = "hoge"
+		sts.Status.CurrentReplicas = 3
+		sts.Status.UpdatedReplicas = 3
+		sts.Status.ObservedGeneration = sts.Generation - 1
+		err = k8sClient.Status().Update(ctx, sts)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("checking condition is false")
+		Eventually(func() error {
+			cluster2 := &mocov1beta2.MySQLCluster{}
+			if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster2); err != nil {
+				return err
+			}
+			conditionUpToDate := meta.FindStatusCondition(cluster2.Status.Conditions, mocov1beta2.ConditionUpToDate)
+			if conditionUpToDate == nil {
+				return fmt.Errorf("not yet updated")
+			}
+			if conditionUpToDate.Status != metav1.ConditionFalse {
+				return fmt.Errorf("not yet updated")
+			}
+			return nil
+		}).Should(Succeed())
+	})
+
+	It("should sets reson to be expected when partition is enabled", func() {
+		cluster := testNewMySQLCluster("test")
+		err := k8sClient.Create(ctx, cluster)
+		Expect(err).NotTo(HaveOccurred())
+
+		var sts *appsv1.StatefulSet
+		Eventually(func() error {
+			sts = &appsv1.StatefulSet{}
+			if err := k8sClient.Get(ctx, client.ObjectKey{Namespace: "test", Name: "moco-test"}, sts); err != nil {
+				return err
+			}
+			return nil
+		}).Should(Succeed())
+
+		By("setting sts status to be partitioned roll out in progress")
+		sts.Spec.UpdateStrategy.Type = appsv1.RollingUpdateStatefulSetStrategyType
+		sts.Spec.UpdateStrategy.RollingUpdate = &appsv1.RollingUpdateStatefulSetStrategy{
+			Partition: pointer.Int32(2),
+		}
+		err = k8sClient.Update(ctx, sts)
+		Expect(err).NotTo(HaveOccurred())
+		sts.Status.Replicas = 3
+		sts.Status.ReadyReplicas = 3
+		sts.Status.AvailableReplicas = 3
+		sts.Status.CurrentRevision = "hoge"
+		sts.Status.UpdateRevision = "fuga"
+		sts.Status.CurrentReplicas = 3
+		sts.Status.UpdatedReplicas = 0
+		sts.Status.ObservedGeneration = sts.Generation
+		err = k8sClient.Status().Update(ctx, sts)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("checking reason is expected")
+		Eventually(func() error {
+			cluster2 := &mocov1beta2.MySQLCluster{}
+			if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(cluster), cluster2); err != nil {
+				return err
+			}
+			conditionUpToDate := meta.FindStatusCondition(cluster2.Status.Conditions, mocov1beta2.ConditionUpToDate)
+			if conditionUpToDate == nil {
+				return fmt.Errorf("not yet updated")
+			}
+			if conditionUpToDate.Message != "UpdateInProgress, Partition is enabled, waiting for partitioned roll out to finish" {
+				return fmt.Errorf("not yet updated: %s", conditionUpToDate.Message)
+			}
+			return nil
+		}).Should(Succeed())
+		cluster = &mocov1beta2.MySQLCluster{}
+		err = k8sClient.Get(ctx, client.ObjectKey{Namespace: "test", Name: "test"}, cluster)
+		Expect(err).NotTo(HaveOccurred())
+		conditionUpToDate := meta.FindStatusCondition(cluster.Status.Conditions, mocov1beta2.ConditionUpToDate)
+		Expect(conditionUpToDate.Status).To(Equal(metav1.ConditionFalse))
```

Comment:
I think it should be checked inside `Eventually`.
This logic can not guarantee that the `UpToDate` is `false` when the message is `UpdateInProgress, Partition is enabled, waiting for partitioned roll out to finish`.

Reply:
fixed.